### ISSUE TITLE
SimpleSAML exposes database password

### ISSUE THREAD

**Issue by nondeterministic** [simplesamlphp/simplesamlphp-module-sqlauth#2]
# **SimpleSAML exposes database password**

**Describe the bug**
When `sqlauth:SQL` is chosen in `authsources.php` and a corresponding PHP-database driver is not yet installed, SimpleSAML throws exception, revealing the `dsn` string - including the database username and password.

**To Reproduce**
1. Install SimpleSAML 1.19.0
2. Use `sqlauth:SQL` in `authsources.php`
3. Make sure that NO PHP-database driver is installed! 
4. Try to authenticate and see "Unhandled exception" message, which includes the `dsn`, e.g.,
```
'pgsql:host=localhost;port=5001;dbname=mydb;user=mydb_user_idp;password=foobar',
```

**Expected behavior**
I would expect that under no circumstances, would SimpleSAML reveal such sensitive information, even in case of an exception. 

**Screenshots or logs**
N/A

**Additional context**
N/A

**Comment by tvdijen** [comment]
I'm actually trying to wrap my head around why you've added the credentials to the dsn...  That's wrong..

**Comment by nondeterministic** [comment]
OK, will try to leave it out and see if that works. The SimpleSAML documentation - https://simplesamlphp.org/docs/stable/sqlauth:sql - refers to the PHP documentation - https://www.php.net/manual/de/ref.pdo-pgsql.connection.php - which states that `dsn` should look as follows: `pgsql:host=localhost;port=5432;dbname=testdb;user=bruce;password=mypass`. And that's what I did.

**Comment by tvdijen** [comment]
Interesting.. It seems this may be different per driver..  I guess it's not too hard to filter it from the exception message..

**Comment by nondeterministic** [comment]
I think, this sums it up. And you're best to know, if you would like to filter it or add a remark in the documentation (or come to the conclusion that I was stupid enough to follow instructions to the letter :-) ) Thanks for your reply anyway!

**Comment by tvdijen** [comment]
We always have to consider the fact that users do stupid things ;), so I think filtering is still a good thing to do

**Comment by tvdijen** [comment]
Keeping this open as a reminder till I find the time to filter it

**Comment by tvdijen** [comment]
Solved in https://github.com/simplesamlphp/simplesamlphp-module-sqlauth/commit/17012e9d31e60cea0802d5b66523ccbc4f62b1a2
Also tagged a new release. You can manually update the module using `composer update` or wait for it to be included in the next v1.19.1 bugfix-release of SimpleSAMLphp
